mteb/abstasks/AbsTaskInstructionRetrieval.py

```diff
@@ -0,0 +1,312 @@
+import logging
+import json
+import os
+from collections import defaultdict
+from time import time
+from typing import Dict, Tuple
+import tqdm
+
+from datasets import load_dataset, Value, Features
+
+from ..evaluation.evaluators import InstructionRetrievalEvaluator
+from ..evaluation.evaluators import utils
+from .AbsTask import AbsTask
+
+logger = logging.getLogger(__name__)
+
+# Adapted from https://github.com/beir-cellar/beir/blob/f062f038c4bfd19a8ca942a9910b1e0d218759d4/beir/datasets/data_loader_hf.py#L10
+class HFDataLoader:    
+    def __init__(self, hf_repo: str = None, hf_repo_qrels: str = None, data_folder: str = None, prefix: str = None, corpus_file: str = "corpus.jsonl", query_file: str = "queries.jsonl", 
+                 qrels_folder: str = "qrels", qrels_file: str = "", streaming: bool = False, keep_in_memory: bool = False):
+        self.corpus = {}
+        self.queries = {}
+        self.qrels = {}
+        self.og_instructions = {}
```

Comment:
The instructions have to be in this file because they come from the underlying data and are not model specific, unlike the prefixes that models currently use as instructions that are model-specific.